src/components/structures/RoomView.js

```diff
@@ -205,6 +205,8 @@ module.exports = React.createClass({
 
         this.setState({
             events: [],
+            searchResults: null, // we may have arrived here by clicking on a
+                                 // search result. Hide the results.
```

Comment:
yeah. The ultimate plan here would be to use something other than RoomView to show the search results.